galleries/tutorials/histogram_normalization.py

```diff
@@ -0,0 +1,373 @@
+# data. The value attached to each bar is divided by the total number of data
+# points *and* the width of the bin, and thus the values *integrate* to one
+# when integrating across the full range of data.
+# e.g. ::
+#
+#     density = counts / (sum(counts) * np.diff(bins))
+#     np.sum(density * np.diff(bins)) == 1
+#
+# This normalization is how `probability density functions
+# <https://en.wikipedia.org/wiki/Probability_density_function>`_ are defined in
+# statistics.  If :math:`X` is a random variable on :math:`x`, then :math:`f_X`
+# is is the probability density function if :math:`P[a<X<b] = \int_a^b f_X dx`.
+# If the units of x are Volts, then the units of :math:`f_X` are :math:`V^{-1}`
+# or probability per change in voltage.
+#
+# The usefulness of this normalization is a little more clear when we draw from
+# a known distribution and try to compare with theory.  So, choose 1000 points
+# from a `normal distribution
+# <https://en.wikipedia.org/wiki/Normal_distribution>`_, and also calculate the
+# known probability density function:
+
+xdata = rng.normal(size=1000)
+xpdf = np.arange(-4, 4, 0.1)
+pdf = 1 / (np.sqrt(2 * np.pi)) * np.exp(-xpdf**2 / 2)
+
+# %%
+# If we don't use ``density=True``, we need to scale the expected probability
+# distribution function by both the length of the data and the width of the
+# bins to fit it to the histogram:
+#
+# .. code-block:: python
+#
+#    ax['False'].hist(xdata, bins=xbins, density=False, histtype='step', label='counts')
+#    ax['False'].plot(xpdf, pdf * len(xdata) * dx,
+#                           label=r'$N\,f_X(x)\,\delta x$', alpha=.5)
+#
+# while we do not need to scale the pdf to fit it to the normalized histogram
+#
+# .. code-block:: python
+#
+#    ax['True'].hist(xdata, bins=xbins, density=True, histtype='step', label='density')
+#    ax['True'].plot(xpdf, pdf, label='$f_X(x)$', alpha=.5)
+#
+# The effect of this scaling is that the two plots look identical, but are on different
+# y-axis scales:
+
+dx = 0.1
+xbins = np.arange(-4, 4, dx)
+
+fig, ax = plt.subplot_mosaic([['False', 'True']], layout='constrained',
+                             figsize=(fw, fh))
+fig.suptitle("Histogram normalization using scaling")
+
+ax['False'].hist(xdata, bins=xbins, density=False, histtype='step', label='counts')
+# scale and plot the expected pdf:
+ax['False'].plot(xpdf, pdf * len(xdata) * dx, label=r'$N\,f_X(x)\,\delta x$', alpha=.5)
+
+
+ax['True'].hist(xdata, bins=xbins, density=True, histtype='step', label='density')
+ax['True'].plot(xpdf, pdf, label='$f_X(x)$', alpha=.5)
+
+
+ax['False'].set(xlabel='x [$V$]', ylabel='Count per bin', title="density=False")
+ax['False'].legend()
+ax['True'].set(xlabel='x [$V$]', ylabel='Probability density [$V^{-1}$]',
+               title="density=True")
+ax['True'].legend()
+
+# %%
+# weights
+# =======
+#
+# Sometimes people want to normalize so that the sum of counts is one.  This is
```

Comment:
This is underselling why the `weight` keyword exists.  If we are going to go with this re-organization the subsections need to be individually complete and correct.

Reply:
I haven't looked at all the changes here, but agreed that the point of the example was not to explain all the uses of the *weight* parameter, but rather to point out that it can be used to get the effect of normalizing by N which people sometimes want.